### ISSUE TITLE
Can´t interact with the device

### ISSUE THREAD

**Issue by Menkaura81** [tud-hri/logitech_steering_wheel#2]
Hello. I´m trying to controll the force feedback of a Logitech Driving Force PRO. With the demo app of the logitech sdk all works fine, but with the wrapped version, i can only initialize the wheel and get device name, manufacturer... none of the interactions with the wheel work. I can set range, but when getting it, it hasn´t changed. Also no ffb effects are played.

**Comment by OlgerSiebinga** [comment]
Hello @manteka, 

This sounds like a familiar problem. It probably has to do with the drivers installed on your computer. Did you try the fix for that as reported in the readme? 

> In some cases, Windows has been known to install other drivers when a steering wheel is plugged in. These default drivers installed by Windows do not work with the SDK. The solution to is problem is to:
> 
> 1. plug in the steering wheel,
> 2. uninstall the driver Windows automatically installed, and leave steering wheel plugged-in and turned on, and
> 3. Install the Logitech Gaming Software (this should automatically install the correct drivers) 

Also, I found it to be quite sensitive to the version of the logitech gaming software that you use 5.10 worked for me. Let me know if this helped.

**Comment by Menkaura81** [comment]
Yeah. Sure it is something related to the drivers. The curious thing is that 5.10 doesn't work for me. Only when i install the current version of the driver it start detecting the wheel. Will do more testing and post my results. Thanks for your response

**Comment by OlgerSiebinga** [comment]
I remember it was also quite sensitive to the order in which you do things. Installing the logitech gaming software and then powering the device did not work for me. I had to install the software while the device was plugged in and powered. but that could also depend on the type of steering wheel you're using.

**Comment by pedroeggs** [comment]
> Hello. I´m trying to controll the force feedback of a Logitech Driving Force PRO. With the demo app of the logitech sdk all works fine, but with the wrapped version, i can only initialize the wheel and get device name, manufacturer... none of the interactions with the wheel work. I can set range, but when getting it, it hasn´t changed. Also no ffb effects are played.

Did you get it to work?
I've been trying on a G29 on Windows 10 and have the same problem as you. Device recognized but no interaction with it.
Went through multiple versions of Logitech Gaming Software and/or Logitech GHUB combinations but none of them worked.

**Comment by OlgerSiebinga** [comment]
Hi @pedroeggs,

Did you try the suggested fix from the readme? Windows installs different drivers that break things. 

> In some cases, Windows has been known to install other drivers when a steering wheel is plugged in. These default drivers 
> installed by Windows do not work with the SDK. The solution to is problem is to:
>
> plug in the steering wheel,
> uninstall the driver Windows automatically installed, and leave steering wheel plugged-in and turned on, and
> Install the Logitech Gaming Software (this should automatically install the correct drivers) 

A second problem is the active window. You need a QT app with a window that is selected for the steering wheel to work.

I hope this helps!

**Comment by pedroeggs** [comment]
Hi @OlgerSiebinga ,

Controlling it via a QT app is what solved it for me. Thanks for clearing that up!

Didn't notice that requirement in any documentation from Logitech so I just didn't think to do it.
The bad side is that I don't even know if I had driver problems since I followed your fix multiple times.

Thanks for respoding so quickly!

**Comment by erfun77** [comment]
> Hi [@pedroeggs](https://github.com/pedroeggs),
> 
> Did you try the suggested fix from the readme? Windows installs different drivers that break things.
> 
> > In some cases, Windows has been known to install other drivers when a steering wheel is plugged in. These default drivers
> > installed by Windows do not work with the SDK. The solution to is problem is to:
> > plug in the steering wheel,
> > uninstall the driver Windows automatically installed, and leave steering wheel plugged-in and turned on, and
> > Install the Logitech Gaming Software (this should automatically install the correct drivers)
> 
> A second problem is the active window. You need a QT app with a window that is selected for the steering wheel to work.
> 
> I hope this helps!

Thank you so much @OlgerSiebinga,

your second solution worked for me. Before that, I literally did everything, but it did not work. You saved my life :)

**Comment by poyta-san** [comment]
It really worked with QT app with a window but it doesn't work if the window is on background. Any tips for me?

**Comment by OlgerSiebinga** [comment]
hi @poyta-san,

Yes, that's correct (unfortunately). As far as I could find, it has to do with how Windows handles user input from USB devices. These are only redirected to a program if the window is active. The rationale behind this is that you want your keyboard (for example) only to interact with the active window and not everything in the background. So, I'm afraid there is no way to deal with it.

If you want to know more I suggest you contact Logitech, as this is related to how they set up lower-level communication with the steering wheels.
